File: Hardware/NodeJs/led-test.js
const controller = require('./output-hardware-controller');

async function wait(ms) {
  return new Promise(resolve => setTimeout(resolve, ms));
}

async function main(){
  let i = 0;
  let j = 0;
  while (true) {
    console.log(`Turning on LED at (${i}, ${j})`);
    await controller.turnOn(0, 0);
    await wait(1000);
    await controller.turnOff(0, 0);

    j = (j + 1) % 3;
    if (j === 0) {
      i = (i + 1) % 2;
    }

  }
}

main();

process.on('SIGINT', () => {
  controller.cleanup();
  console.log("Clean exit.");
  process.exit();
});

File: Hardware/Python/LedTest.py
import asyncio
import signal
import sys
import LedControll_Demo as controller

async def wait(ms):
    """Wait for specified milliseconds"""
    await asyncio.sleep(ms / 1000)

async def main():
    i = 0
    while True:
        print(f"Turning on LED at ({i})")
        controller.turn_on(i)
        await wait(1000)
        controller.turn_off(i)
        i = (i + 1) % 4
        
def cleanup(signum=None, frame=None):
    """Cleanup on exit"""
    print("Clean exit.")
    controller.cleanup()
    sys.exit(0)

# Register signal handlers
signal.signal(signal.SIGINT, cleanup)

if __name__ == "__main__":
    try:
        # Run the async main function
        asyncio.run(main())
    except KeyboardInterrupt:
        cleanup()